### DISCUSSION TITLE
Projection mode in rendering.py

### DISCUSSION THREAD

**Discussion by Yarkane** [jimy-byerley/pymadcad#42] [Q&A] [answered]
Hello there.

We started to implement orthographic projection in our project. From what I understand, the Orthographic mode of projection has to be defined when initializing the View object.

![image](https://user-images.githubusercontent.com/45920565/181245152-30af3334-2c15-49d4-a135-1c132120b538.png)

But as we can see in this screen, "projection" is not used in the function. I edited the line
`ViewCommon.__init__(self, scene, projection=None, navigation=None)`
to
`ViewCommon.__init__(self, scene, projection=projection, navigation=None)`
and called the function giving as an argument a `madcad.rendering.Orthographic()` object.
which resumed to what I expected : an orthographic view.

![image](https://user-images.githubusercontent.com/45920565/181246500-3908f2bc-464e-4d20-b3f7-75d68a3ba3ba.png)

Why is it necessary to change the code in rendering.py ? Is there a reason why it has been "disabled" in the code ?

Thanks in advance !

**Comment by jimy-byerley** [comment, marked as answer]
No reason at all, I think I just made a wrong copy-paste at some moment I was refactoring `ViewCommon` and never noticed these parameters were not working anymore because I personally always use the defaults from the configuration.
Thanks for noticing me ! I will change that in the next release.

I should however check that all the methods of `View` still work with the orthographic projection. I didn't tested it in the past year, so other things might be broken too :-/

**Comment by Yarkane** [comment]
Thanks a lot for the answer ! Keep me in touch as possible :)

**Reply by jimy-byerley** [reply]
Finally the fix is released in [v0.13](https://github.com/jimy-byerley/pymadcad/releases/tag/v0.13), in addition with a rework of `Orthographic` projection, so I'm sure it will work starting from now ;)
